windows/src/engine/keyman32/appint/appint.cpp

```diff
@@ -203,3 +203,37 @@ AppIntegration::AppIntegration()
 	hwnd = NULL;
   FShiftFlags = 0;
 }
+
+BOOL ContextItemsFromAppContext(WCHAR const* buf, km_kbp_context_item** outPtr)
+{
+
+  km_kbp_context_item* context_items  = new km_kbp_context_item[wcslen(buf) + 1];
+  // would like to check bad allocation but nothrow not known to this compiler
+  WCHAR const *p = buf;
+  uint8_t contextIndex = 0;
+  while (*p) {
+    if (*p == UC_SENTINEL) {
+      // we know the only uc_sentinel code in the context is code_deadkey, which has only 1 parameter: uc_sentinel code_deadkey <deadkey_id>
+      // setup dead key context item
+      p += 2;
+      context_items[contextIndex++] = km_kbp_context_item{ KM_KBP_CT_MARKER, {0,}, {*p} };
+    }
+    else {  // setup character need to handle surrogate
```

Comment:
It would be better to do this as `} else if {` to remove an indent level.

Reply:
done